qiskit/utils/quantum_instance.py

```diff
@@ -124,15 +124,24 @@ def __init__(self,
         self._pass_manager = pass_manager
 
         # setup run config
-        if shots is not None:
-            if self.is_statevector and shots != 1:
-                logger.info("statevector backend only works with shot=1, changing "
-                            "shots from %s to 1.", shots)
-                shots = 1
-            max_shots = self._backend.configuration().max_shots
-            if max_shots is not None and shots > max_shots:
-                raise QiskitError('The maximum shots supported by the selected backend is {} '
-                                  'but you specified {}'.format(max_shots, shots))
+        if self.is_statevector and shots != 1:
+            logger.info("statevector backend only works with shot=1, changing "
+                        "shots from %s to 1.", shots)
+            shots = 1
+
+        # TODO: exchange this for a Backend instancecheck once Aer's simulators implement that
+        # interface
+        if hasattr(backend, 'options'):
+            if 'shots' in backend.options:
+                if shots != backend.options['shots']:
+                    logger.info('Overwriting the number of shots in the quantum instance with the '
+                                'settings from the backend.')
+                shots = backend.options.get('shots', 1024)
```

Comment:
Shouldn't we add to the `shots` docstring in the constructor that if the user sets the number of shots in the backend that this will override any value given directly to the QuantumInstance?